### PR TITLE
8339918: Remove checks for outdated -t -tm -Xfuture -checksource -cs -noasyncgc options from the launcher

### PR DESCRIPTION
**Pull request by jaikiran**
Can I please get a review for this change which cleans up the `java` launcher to remove checks/support for outdated options?

As noted in https://bugs.openjdk.org/browse/JDK-8339918, these 6 options have been outdated and unsupported for several releases now. 2 of them even throw an error currently. The change in this PR removes the code which had specific checks for these options and will now consider these options just like any other unknown option to the launcher.

tier1, tier2 and tier3 testing passed with these changes. Higher tier testing is in progress.

Would this change require a CSR?

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8339928](https://bugs.openjdk.org/browse/JDK-8339928) to be approved

### Issues
 * [JDK-8339918](https://bugs.openjdk.org/browse/JDK-8339918): Remove checks for outdated -t -tm -Xfuture -checksource -cs -noasyncgc options from the launcher (**Sub-task** - P4)
 * [JDK-8339928](https://bugs.openjdk.org/browse/JDK-8339928): Remove checks for outdated -t -tm -Xfuture -checksource -cs -noasyncgc options from the launcher (**CSR**)

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20945/head:pull/20945` \
`$ git checkout pull/20945`

Update a local copy of the PR: \
`$ git checkout pull/20945` \
`$ git pull https://git.openjdk.org/jdk.git pull/20945/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20945`

View PR using the GUI difftool: \
`$ git pr show -t 20945`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20945.diff">https://git.openjdk.org/jdk/pull/20945.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20945#issuecomment-2343175664)

### CODE REVIEW

**Comment by AlanBateman** [comment]
> Would this change require a CSR?

and probably a release note too.

**Comment by jaikiran** [comment]
The CSR for this change is now ready for review at https://bugs.openjdk.org/browse/JDK-8339928

**Review by dholmes-ora — APPROVED** [review]
Good cleanup.

Thanks

**Review by AlanBateman — APPROVED** [review]

**Comment by ecki** [comment]
What about `-Xdebug` it’s deprecated for remove. It’s not yet moved to the obsolete section of the manual. And we could also remove it from the -X command line help.

File: src/java.base/share/man/java.1 [hunk 1]
```diff
@@ -3920,6 +3914,16 @@ of objects reachable from the old generation space into the young
 generation space.
 To disable GC of the young generation before each full GC, specify the
 option \f[V]-XX:-ScavengeBeforeFullGC\f[R].
+.TP
+\f[V]-Xfuture\f[R]
```
Comment (ecki) [inline src/java.base/share/man/java.1:3918]: Maybe also remove the -X help (or at least change its description) in https://github.com/openjdk/jdk/blob/c91fa278fe17ab204beef0fcef1ada6dd0bc37bb/src/java.base/share/classes/sun/launcher/resources/launcher.properties#L144

**Comment by ecki** [comment]
BTW somewhat related, I always wondered if `-Xdiag` is still needed or can be better documented. For example it does not describe that it mainly modified the launcher diagnostic messages (and very little of them)

**Comment by dholmes-ora** [comment]
> What about `-Xdebug` it’s deprecated for remove. It’s not yet moved to the obsolete section of the manual. And we could also remove it from the -X command line help.

If it is deprecated then it goes in the Deprecated section. The obsoleted section doesn't apply to -X flags but is part of the three-stage removal for Hotspot -XX flags.

File: src/java.base/share/man/java.1 (hunk 1 above)
Comment (dholmes-ora) [reply src/java.base/share/man/java.1:3918]: Yes this needs to be removed from the help text.

Comment (jaikiran) [reply src/java.base/share/man/java.1:3918]: Thank you for spotting that Bernd. I have updated the PR to remove that text as well as updated an example program that was using `-noasyncgc`. I have searched the code for other similar occurences and haven't found anymore.

**Review by dholmes-ora — APPROVED** [review]
Looks good. But another cleanup is also possible

File: src/jdk.jdi/share/classes/com/sun/tools/example/debug/tty/TTY.java [hunk 2]
```diff
@@ -988,7 +988,7 @@ public static void main(String argv[]) throws MissingResourceException {
                    token.startsWith("-X") ||
                    // Old-style options (These should remain in place as long as
                    //  the standard VM accepts them)
-                   token.equals("-noasyncgc") || token.equals("-prof") ||
+                   token.equals("-prof") ||
```
[the remark this answers is not in the document]
Comment (jaikiran) [reply src/jdk.jdi/share/classes/com/sun/tools/example/debug/tty/TTY.java]: Done. There also was `-v` and `-v:...` in this example which aren't supported either (not even in JDK 8). So I've removed those as well.

[the remark this answers is not in the document]
Comment (dholmes-ora) [reply src/jdk.jdi/share/classes/com/sun/tools/example/debug/tty/TTY.java]: What about:
```
                 token.startsWith("-ms") || token.startsWith("-mx") ||
                   token.startsWith("-ss") || token.startsWith("-oss") ) {
```
? `-oss` is gone as of 9. The others are not need as the -X counterpart can be used.

[the remark this answers is not in the document]
Comment (jaikiran) [reply src/jdk.jdi/share/classes/com/sun/tools/example/debug/tty/TTY.java]: One of my next plans, as part of the launcher options cleanup https://bugs.openjdk.org/browse/JDK-8286851, was to create a task to deprecate for removal -ms and -mx. I was thinking of removing these usage in this code, as part of that effort. But yes, I can remove these options from this example here as part of this PR - I'll go ahead and do it now.

**Review by dholmes-ora — APPROVED** [review]
Ship it! :)

**Review by AlanBateman — APPROVED** [review]

**Comment by jaikiran** [comment]
Thank you Alan and David for the reviews.

**Pull request closed without merging**